src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -1303,14 +1279,31 @@ private bool InternalSetTaskParameter(TaskPropertyInfo parameter, IList paramete
 
             if (LogTaskInputs && !_taskLoggingContext.LoggingService.OnlyLogCriticalEvents)
             {
-                // If the type is a list, we already logged the parameters
-                if (!(parameterValue is IList))
+                IList parameterValueAsList = parameterValue as IList;
+                bool legacyBehavior = !ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_10);
```

Comment:
This will go into 17.11 so I'll introduce a new changewave and update this check to use it.

Reply:
Is it needed? I thought we create new changeways only for LTS releases (so next one would be 17.12)

Reply:
Yes, the new changewave should be labeled 17.12. I don't think 17.12 has been announced (and 16 made it only to 16.11 :smile:) but 17.11 is non-LTS so it's a reasonable assumption.

Reply:
Ooooh - I thought we accumulate untill the next LTS.
But we actually should create one as soon as we are done with current LTS (even though next one is not yet known). Now I see we have it properly documented - https://github.com/dotnet/msbuild/blob/main/documentation/wiki/ChangeWaves-Dev.md#change-wave-versioning - not sure why I was thinking we wait until we start working on the next LTS..

FYI @f-alizada - as I misinformed you on this one. You should move your change under 17.12 as well - https://github.com/dotnet/msbuild/pull/9874/files#diff-854a463890211b66f6c60646fe5d85736e82b0daaa2152561a055b333db480a8R38 - Sorry for telling you otherwise!